### PR TITLE
[SR SQL Planner Summer Camp] Support cancel alter in new parser

### PR DESCRIPTION
**Pull request by xlwh**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7938

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 26 / 33 (78.79%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CancelAlterTableStmt.java|2|5|40.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/CancelAlterTableStatementAnalyzer.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|11|12|91.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|8|8|100.00%|

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 26 / 33 (78.79%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CancelAlterTableStmt.java|2|5|40.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/CancelAlterTableStatementAnalyzer.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|11|12|91.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|8|8|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeAlterTableStatementTest.java [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
```
Comment (HangyuanLiu) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeAlterTableStatementTest.java:1]: Cannot add StarRocks license to old files

Comment (imay) [reply fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeAlterTableStatementTest.java:1]: this file is created by us. so I think it is OK.

Comment (xlwh) [reply fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeAlterTableStatementTest.java:1]: Yes, it looks like we added it, should use Elastic License 2.0

**Review by HangyuanLiu — DISMISSED** [review]

**Review by imay — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by xlwh** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 27 / 33 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[122]|
|:large_blue_circle:|com/starrocks/analysis/CancelAlterTableStmt.java|2|5|40.00%|[55, 56, 104]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CancelAlterTableStatementAnalyzer.java|3|5|60.00%|[11, 17]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|8|8|100.00%|[]|

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 27 / 33 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|[130]|
|:large_blue_circle:|com/starrocks/analysis/CancelAlterTableStmt.java|2|5|40.00%|[55, 56, 104]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CancelAlterTableStatementAnalyzer.java|3|5|60.00%|[11, 17]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|8|8|100.00%|[]|

**Pull request merged**